### ISSUE TITLE
Dart -> Engine roll stopped on `dart fix` issues (?)

### ISSUE THREAD

**Issue by zanderso** [flutter/flutter#154413]
The Engine -> framework roll started failing on the roll here: https://github.com/flutter/flutter/pull/154388

Containing only the Dart -> Engine roll here: https://github.com/flutter/engine/pull/54886

With commits: https://dart.googlesource.com/sdk.git/+log/139867e57ba9..88d311c9a531

Failures are as in https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8738200558966773761/+/u/run_test.dart_for_framework_tests_shard_and_subshard_misc/stdout

and

https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8738200558966773777/+/u/run_test.dart_for_framework_tests_shard_and_subshard_slow/stdout

Eyeballing those, it looks like https://dart.googlesource.com/sdk.git/+/49c6679e4a7392eefee88c55392a0d805afa191a might be related. cc @scheglov

**Comment by scheglov** [comment]
I reverted this CL in https://dart-review.googlesource.com/c/sdk/+/382903

**Comment by scheglov** [comment]
I'm looking now at
https://github.com/flutter/flutter/blob/620d1ea7295ea7244c8fcc5b57ac4e4c7252c647/dev/bots/test/analyze-snippet-code-test-input/known_broken_documentation.dart#L146-L149

And this change was actually intentional in the analyzer.
In the presence of unresolved classes we cannot claim that the import is unused.
It might be unused, but it could also be a typo in the unresolved name, and the import will become used.

How to roll forward these changes?
I cannot update existing expectations because the analyzer used currently will report the warning.
And I cannot land the analyzer changes because existing expectations will change.

**Comment by zanderso** [comment]
In the past we've handled this with manually shepherded rolls as in https://github.com/flutter/flutter/commit/982e00527cff475f152fa97e14a02c82ec527290, but I wonder if @goderbauer would be okay with a different process like disabling the one failing example, rolling in the change, then re-enabling the test with the fix. I'm also curious if there are any thoughts on how to make this a soft breaking change rather than a hard breaking change.

**Comment by goderbauer** [comment]
I am ok with disabling this test to roll that change in and then I am happy to adjust the test to the new reality.

**Comment by scheglov** [comment]
I re-landed the change as https://dart-review.googlesource.com/c/sdk/+/383022
So, now we have the same failing test in Flutter.
Is there anything I should do to fix this?

I tried quickly, but cannot actually reproduce the same state, because to do this I need somehow convince Flutter repository use my version of Dart SDK, and run tests with it, not with the cached version. And I don't know how to do this. But I can try to do this blindly, just remove the `unused_import` line from the expectations.

**Comment by zanderso** [comment]
@scheglov Apologies if the suggestion was unclear, but the idea was to disable the test in the framework before re-landing the change in the Dart SDK.

**Comment by scheglov** [comment]
Oh, sorry, I probably misunderstood it. I looked at the previous change and it seemed that a fix was made together with SDK roll, or something like this.

Do you want me to revert the Dart SDK change for now?

**Comment by zanderso** [comment]
Yeah, reverting to unblock the rolls would be helpful. The Dart roll is very backed up. The monorepo builder is also red https://ci.chromium.org/p/dart/g/monorepo/console.

**Comment by scheglov** [comment]
OK, revert is in progress: https://dart-review.googlesource.com/c/sdk/+/383380

I will rework https://github.com/flutter/flutter/pull/154514 then to `skip` the failing tests.
Although I will still do it blindly, because I cannot actually test it with the Dart SDK changes.

**Comment by a-siva** [comment]
Looks like we have a roll of Dart with the above revert into Flutter, closing issue.
